pkg/api/file_repository.go

```diff
@@ -0,0 +1,99 @@
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// FileSystemRepository represents a mongodb repository
+type FileSystemRepository struct {
+	definitions map[string]*Definition
+}
+
+// NewFileSystemRepository creates a mongo country repo
+func NewFileSystemRepository(dir string) (*FileSystemRepository, error) {
+	repo := &FileSystemRepository{make(map[string]*Definition)}
+	// Grab json files from directory
+	files, err := ioutil.ReadDir(dir)
+	if nil != err {
+		return nil, err
+	}
+
+	for _, f := range files {
+		if strings.Contains(f.Name(), ".json") {
+			filePath := filepath.Join(dir, f.Name())
+			log.Info("Loading API Specification from ", filePath)
+			appConfigBody, err := ioutil.ReadFile(filePath)
+			if err != nil {
+				log.WithError(err).Error("Couldn't load app configuration file")
```

Comment:
`log.WithError(err).WithField("path", filePath).Error("Couldn't load app configuration file")` - we need to have path in error as log level can be set to `error` on production environment